File: jdbc-examples/src/main/js/io/vertx/example/jdbc/streaming/jdbc_example.js
var JDBCClient = require("vertx-jdbc-js/jdbc_client");

var client = JDBCClient.createShared(vertx, {
  "url" : "jdbc:hsqldb:mem:test?shutdown=true",
  "driver_class" : "org.hsqldb.jdbcDriver",
  "max_pool_size" : 30
});

client.getConnection(function (conn, conn_err) {
  if (conn_err != null) {
    console.error(conn_err.getMessage());
    return
  }

  var connection = conn;
  connection.execute("create table test(id int primary key, name varchar(255))", function (res, res_err) {
    if (res_err != null) {
      throw res_err;
    }
    // insert some test data
    connection.execute("insert into test values (1, 'Hello'), (2, 'Goodbye'), (3, 'Cya Later')", function (insert, insert_err) {
      // query some data
      connection.queryStream("select * from test", function (stream, stream_err) {
        if (stream_err == null) {
          var sqlRowStream = stream;

          sqlRowStream.handler(function (row) {
            // do something with the row...
            console.log(JSON.stringify(row));
          }).endHandler(function (v) {
            // no more data available, close the connection
            connection.close(function (done, done_err) {
              if (done_err != null) {
                throw done_err;
              }
            });
          });
        }
      });
    });
  });
});

File: jdbc-examples/src/main/ruby/io/vertx/example/jdbc/streaming/jdbc_example.rb
require 'json'
require 'vertx-jdbc/jdbc_client'

client = VertxJdbc::JDBCClient.create_shared($vertx, {
  'url' => "jdbc:hsqldb:mem:test?shutdown=true",
  'driver_class' => "org.hsqldb.jdbcDriver",
  'max_pool_size' => 30,
  'user' => "SA",
  'password' => ""
})

client.get_connection() { |conn_err,conn|
  if (conn_err != nil)
    STDERR.puts conn_err.get_message()
    return
  end

  connection = conn
  connection.execute("create table test(id int primary key, name varchar(255))") { |res_err,res|
    if (res_err != nil)
      raise res_err
    end
    # insert some test data
    connection.execute("insert into test values (1, 'Hello'), (2, 'Goodbye'), (3, 'Cya Later')") { |insert_err,insert|
      # query some data
      connection.query_stream("select * from test") { |stream_err,stream|
        if (stream_err == nil)
          sqlRowStream = stream

          sqlRowStream.handler() { |row|
            # do something with the row...
            puts JSON.generate(row)
          }.end_handler() { |v|
            # no more data available, close the connection
            connection.close() { |done_err,done|
              if (done_err != nil)
                raise done_err
              end
            }
          }
        end
      }
    }
  }
}
